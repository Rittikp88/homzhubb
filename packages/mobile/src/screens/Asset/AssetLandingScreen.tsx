import React from 'react';
import { StyleSheet, View, Image } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { StorageService, StorageKeys } from '@homzhub/common/src/services/storage/StorageService';
import { IRefreshTokenPayload, IUserPayload } from '@homzhub/common/src/domain/repositories/interfaces';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { UserActions } from '@homzhub/common/src/modules/user/actions';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { theme } from '@homzhub/common/src/styles/theme';
import { images } from '@homzhub/common/src/assets/images';
import { Text, Label, Button, SVGUri } from '@homzhub/common/src/components';
import { GradientBackground } from '@homzhub/mobile/src/components';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { AppStackParamList } from '@homzhub/mobile/src/navigation/AppNavigator';
import { User } from '@homzhub/common/src/domain/models/User';

interface IStateProps {
  user: User | null;
}

interface IDispatchProps {
  logout: (data: IRefreshTokenPayload) => void;
}

type libraryProps = WithTranslation & NavigationScreenProps<AppStackParamList, ScreensKeys.PropertyPostLandingScreen>;
type Props = IStateProps & IDispatchProps & libraryProps;

export class AssetLandingScreen extends React.PureComponent<Props, {}> {
  public render(): React.ReactNode {
    const { t, user } = this.props;
    return (
      <GradientBackground>
        <View style={styles.container}>
          <View style={styles.imagesContainer}>
            <Image source={images.landingScreenLogo} />
            <Text type="regular" textType="semiBold">
              {t('welcomeUser', { username: user?.fullName })}
            </Text>
            <Text type="small" textType="regular" style={styles.description}>
              {t('description')}
            </Text>
            <SVGUri
              width={295}
              height={133}
              viewBox="0 0 295 133"
              preserveAspectRatio="xMidYMid meet"
              uri="https://homzhub-bucket.s3.amazonaws.com/Group+1168.svg"
            />
            <Label type="regular" textType="regular">
              {t('searchProject')}
            </Label>
          </View>
          <View style={styles.buttonContainer}>
            <Button
              title={t('addProperty')}
              type="primary"
              onPress={this.onAddProperty}
              containerStyle={styles.addProperty}
              testID="btnAddProperty"
            />
            {/* TODO: To be removed once the logic is in place */}
            <Button
              title="Dashboard"
              type="secondary"
              onPress={this.onSwitchProperty}
              containerStyle={styles.addProperty}
            />
            <Label type="large" textType="regular" style={styles.logoutContainer}>
              {t('logoutHelperText')} &nbsp;
              <Label type="large" textType="bold" style={styles.logout} onPress={this.logout} testID="lblLogout">
                {t('common:logout')}
              </Label>
            </Label>
          </View>
        </View>
      </GradientBackground>
    );
  }

  public onAddProperty = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.PropertyPostStack, { screen: ScreensKeys.AssetLocationSearch });
  };

  public onSwitchProperty = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.BottomTabs, { screen: ScreensKeys.Dashboard });
  };

  public logout = async (): Promise<void> => {
    const { logout } = this.props;
    const user: IUserPayload | null = (await StorageService.get(StorageKeys.USER)) ?? null;
    if (!user) {
      return;
    }
    const { refresh_token } = user;
    const logoutPayload = {
      refresh_token,
    };
    logout(logoutPayload);
  };
}

export const mapStateToProps = (state: IState): IStateProps => {
  return {
    user: UserSelector.getUserDetails(state),
  };
};

export const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { logout } = UserActions;
  return bindActionCreators(
    {
      logout,
    },
    dispatch
  );
};

export default connect<IStateProps, IDispatchProps, WithTranslation, IState>(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation(LocaleConstants.namespacesKey.property)(AssetLandingScreen));

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.colors.white,
    shadowColor: theme.colors.shadow,
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowOpacity: 0.51,
    shadowRadius: 13.16,
    elevation: 20,
    width: theme.viewport.width,
    maxHeight: theme.viewport.height - 100,
    borderRadius: 8,
    margin: theme.layout.screenPadding,
    paddingHorizontal: theme.layout.screenPadding,
  },
  imagesContainer: {
    justifyContent: 'space-around',
    alignItems: 'center',
    flex: 0.9,
  },
  description: {
    textAlign: 'center',
  },
  buttonContainer: {
    justifyContent: 'space-around',
    alignItems: 'stretch',
    marginVertical: 20,
  },
  addProperty: {
    flex: 0,
    marginVertical: 5,
  },
  logoutContainer: {
    alignSelf: 'center',
    marginVertical: 10,
  },
  logout: {
    color: theme.colors.blue,
  },
});
