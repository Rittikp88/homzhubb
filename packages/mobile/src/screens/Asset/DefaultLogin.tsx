import React from 'react';
import { View, StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { CommonActions } from '@react-navigation/native';
import { theme } from '@homzhub/common/src/styles/theme';
import Logo from '@homzhub/common/src/assets/images/homzhubLogo.svg';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { BottomTabNavigatorParamList } from '@homzhub/mobile/src/navigation/BottomTabs';
import { Button } from '@homzhub/common/src/components/atoms/Button';
import { Text } from '@homzhub/common/src/components/atoms/Text';

type libraryProps = WithTranslation & NavigationScreenProps<BottomTabNavigatorParamList, ScreensKeys.DefaultLogin>;
type IProps = libraryProps;

export class DefaultLogin extends React.PureComponent<IProps> {
  public render(): React.ReactElement {
    const { t } = this.props;
    return (
      <View style={styles.container}>
        <Logo height={100} width={100} />
        <Text type="small" textType="regular" style={styles.loginText}>
          {t('pleaseSignup')}
        </Text>
        <Button type="primary" title={t('signUp')} containerStyle={styles.button} onPress={this.navigateToSignup} />
      </View>
    );
  }

  public navigateToSignup = (): void => {
    const { navigation } = this.props;
    navigation.dispatch(
      CommonActions.navigate({
        name: ScreensKeys.AuthStack,
      })
    );
  };
}

export default withTranslation()(DefaultLogin);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.colors.white,
  },
  loginText: {
    marginVertical: 10,
  },
  button: {
    flex: 0,
    marginVertical: 10,
  },
});
