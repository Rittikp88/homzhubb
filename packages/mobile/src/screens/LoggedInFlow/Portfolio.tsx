import React from 'react';
import { View, StyleSheet, Text } from 'react-native';

export class Portfolio extends React.PureComponent<{}, {}> {
  public render = (): React.ReactElement => {
    return (
      <View style={styles.screen}>
        <Text>Portfolio Screen</Text>
      </View>
    );
  };
}

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
