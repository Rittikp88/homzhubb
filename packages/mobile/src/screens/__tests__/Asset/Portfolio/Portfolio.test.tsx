// @ts-nocheck
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Portfolio } from '@homzhub/mobile/src/screens/Asset/Portfolio';
import { PortfolioAssetData, TenanciesAssetData } from '@homzhub/common/src/mocks/AssetData';

const mock = jest.fn();

describe('Portfolio Screen', () => {
  let component: ShallowWrapper;
  let props: any;

  beforeEach(() => {
    props = {
      navigation: {
        navigate: mock,
      },
    };
    component = shallow(<Portfolio {...props} t={(key: string): string => key} />);
  });

  it('should render snapshot', () => {
    component.instance().updateFilter();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for isFullScreenView', () => {
    component.instance().onFullScreenToggle();
    component.instance().updateSlide(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for tenancies', () => {
    component.setState({ tenancies: TenanciesAssetData });
    component.instance().renderList(TenanciesAssetData[0]);
    component.instance().handleExpandCollapse(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for properties', () => {
    component.setState({ portfolioProperties: PortfolioAssetData });
    component.instance().onViewProperty();
    component.instance().onSelectFilter('ALL');
    expect(toJson(component)).toMatchSnapshot();
  });
});
