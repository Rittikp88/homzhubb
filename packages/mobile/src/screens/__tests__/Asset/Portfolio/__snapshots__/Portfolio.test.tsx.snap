// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Screen should render snapshot 1`] = `
<AnimatedProfileHeader
  title="Portfolio"
>
  <AssetMetricsList
    assetCount={10}
    containerStyle={
      Object {
        "marginVertical": 12,
      }
    }
    data={
      Array [
        Object {
          "header": "Residential",
          "id": 1,
          "value": 5,
        },
        Object {
          "header": "Commercial",
          "id": 2,
          "value": 3,
        },
        Object {
          "header": "Corporate",
          "id": 3,
          "value": 2,
        },
      ]
    }
    isPortfolio={true}
    subscription="HomzHub Pro"
  />
  <Text
    style={
      Object {
        "color": "#001B36",
        "marginBottom": 16,
        "marginTop": 4,
      }
    }
    textType="semiBold"
    type="small"
  >
    Tenancies
  </Text>
  <FlatList
    data={
      Array [
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#CE9B6C",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "OWNER",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#CE9B6C",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "OWNER",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#001B36",
          "marginBottom": 16,
          "marginTop": 4,
        }
      }
      textType="semiBold"
      type="small"
    >
      Property Portfolio
    </Text>
    <Icon
      allowFontScaling={false}
      color="#4C5F72"
      name="vertical-dots"
      onPress={[Function]}
      size={18}
      testID="menu"
    />
  </View>
  <FlatList
    data={
      Array [
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#CE9B6C",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "OWNER",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#6CCEC1",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "RENTED",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#FF8576",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 3,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "VACANT",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#FF8576",
          "id": 4,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": false,
          "property_name": "2BHK - Godrej Prime",
          "type": "VACANT",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#FBC02D",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 5,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "FOR SALE",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#58C2f1",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 6,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "MAINTAIN",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <BottomSheetListView
    data={
      Array [
        Object {
          "label": "All",
          "value": "all",
        },
        Object {
          "label": "Rented",
          "value": "rented",
        },
        Object {
          "label": "Vacant",
          "value": "vacant",
        },
        Object {
          "label": "For Sale",
          "value": "sale",
        },
        Object {
          "label": "Maintain",
          "value": "maintain",
        },
      ]
    }
    isBottomSheetVisible={false}
    listHeight={500}
    listTitle="Filter"
    onCloseDropDown={[Function]}
    onSelectItem={[Function]}
    selectedValue="all"
  />
</AnimatedProfileHeader>
`;

exports[`Portfolio Screen should render snapshot for functions 1`] = `
<AnimatedProfileHeader
  title="Portfolio"
>
  <AssetMetricsList
    assetCount={10}
    containerStyle={
      Object {
        "marginVertical": 12,
      }
    }
    data={
      Array [
        Object {
          "header": "Residential",
          "id": 1,
          "value": 5,
        },
        Object {
          "header": "Commercial",
          "id": 2,
          "value": 3,
        },
        Object {
          "header": "Corporate",
          "id": 3,
          "value": 2,
        },
      ]
    }
    isPortfolio={true}
    subscription="HomzHub Pro"
  />
  <Text
    style={
      Object {
        "color": "#001B36",
        "marginBottom": 16,
        "marginTop": 4,
      }
    }
    textType="semiBold"
    type="small"
  >
    Tenancies
  </Text>
  <FlatList
    data={
      Array [
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#CE9B6C",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "OWNER",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#CE9B6C",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "OWNER",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#001B36",
          "marginBottom": 16,
          "marginTop": 4,
        }
      }
      textType="semiBold"
      type="small"
    >
      Property Portfolio
    </Text>
    <Icon
      allowFontScaling={false}
      color="#4C5F72"
      name="vertical-dots"
      onPress={[Function]}
      size={18}
      testID="menu"
    />
  </View>
  <FlatList
    data={
      Array [
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#CE9B6C",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "OWNER",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#6CCEC1",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "RENTED",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#FF8576",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 3,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "VACANT",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#FF8576",
          "id": 4,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": false,
          "property_name": "2BHK - Godrej Prime",
          "type": "VACANT",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#FBC02D",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 5,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "FOR SALE",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#58C2f1",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 6,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "MAINTAIN",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <BottomSheetListView
    data={
      Array [
        Object {
          "label": "All",
          "value": "all",
        },
        Object {
          "label": "Rented",
          "value": "rented",
        },
        Object {
          "label": "Vacant",
          "value": "vacant",
        },
        Object {
          "label": "For Sale",
          "value": "sale",
        },
        Object {
          "label": "Maintain",
          "value": "maintain",
        },
      ]
    }
    isBottomSheetVisible={false}
    listHeight={500}
    listTitle="Filter"
    onCloseDropDown={[Function]}
    onSelectItem={[Function]}
    selectedValue="rent"
  />
</AnimatedProfileHeader>
`;

exports[`Portfolio Screen should render snapshot for state 1`] = `
<AnimatedProfileHeader
  title="Portfolio"
>
  <AssetMetricsList
    assetCount={10}
    containerStyle={
      Object {
        "marginVertical": 12,
      }
    }
    data={
      Array [
        Object {
          "header": "Residential",
          "id": 1,
          "value": 5,
        },
        Object {
          "header": "Commercial",
          "id": 2,
          "value": 3,
        },
        Object {
          "header": "Corporate",
          "id": 3,
          "value": 2,
        },
      ]
    }
    isPortfolio={true}
    subscription="HomzHub Pro"
  />
  <Text
    style={
      Object {
        "color": "#001B36",
        "marginBottom": 16,
        "marginTop": 4,
      }
    }
    textType="semiBold"
    type="small"
  >
    Tenancies
  </Text>
  <FlatList
    data={
      Array [
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#CE9B6C",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "OWNER",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#CE9B6C",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "OWNER",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#001B36",
          "marginBottom": 16,
          "marginTop": 4,
        }
      }
      textType="semiBold"
      type="small"
    >
      Property Portfolio
    </Text>
    <Icon
      allowFontScaling={false}
      color="#4C5F72"
      name="vertical-dots"
      onPress={[Function]}
      size={18}
      testID="menu"
    />
  </View>
  <FlatList
    data={
      Array [
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#CE9B6C",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "OWNER",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#6CCEC1",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "RENTED",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#FF8576",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 3,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "VACANT",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#FF8576",
          "id": 4,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": false,
          "property_name": "2BHK - Godrej Prime",
          "type": "VACANT",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#FBC02D",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 5,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "FOR SALE",
        },
        Object {
          "address": "Sindhi Society, Chembur, Mumbai- 400071",
          "color": "#58C2f1",
          "contacts": Object {
            "country_code": "+91",
            "email": "anuvrat.somnath@nineleaps.com",
            "full_name": "Anuvrat Somnath",
            "id": 1,
            "phone_number": "7903470293",
          },
          "id": 6,
          "images": Array [
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": Object {
                "testUri": "../../../packages/common/src/assets/images/property.png",
              },
            },
          ],
          "isPropertyCompleted": true,
          "property_name": "2BHK - Godrej Prime",
          "type": "MAINTAIN",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <BottomSheetListView
    data={
      Array [
        Object {
          "label": "All",
          "value": "all",
        },
        Object {
          "label": "Rented",
          "value": "rented",
        },
        Object {
          "label": "Vacant",
          "value": "vacant",
        },
        Object {
          "label": "For Sale",
          "value": "sale",
        },
        Object {
          "label": "Maintain",
          "value": "maintain",
        },
      ]
    }
    isBottomSheetVisible={true}
    listHeight={500}
    listTitle="Filter"
    onCloseDropDown={[Function]}
    onSelectItem={[Function]}
    selectedValue="all"
  />
</AnimatedProfileHeader>
`;
